### PR TITLE
alternative fix to mypy errors

### PR DESCRIPTION
**Pull request by siv2r**
Here is an alternative fix for mypy errors without using the `#type: ignore` comment.

Changes:
- explicitly check for infinity point using `is None` instead of `is_infinity()`
- check for the infinity point before passing the variable to `compress_point()`

### CODE REVIEW

File: reference.py [hunk 1]
```diff
@@ -142,11 +142,11 @@ def schnorr_presig_sign(msg: bytes, seckey: bytes, aux_rand: bytes, T: bytes) ->
     assert R is not None
     T_point = cpoint(T)
     R0 = point_add(R, T_point) # elliptic curve point R0 = R + T
-    if is_infinite(R0):
+    if R0 is None: # fail if infinity point
```
Comment (jonasnick) [inline reference.py]:
```suggestion
    if R0 is None: # fail if point at infinity
```

**Review by jonasnick — COMMENTED** [review]
Concept ACK

File: reference.py (hunk 1 above)
Comment (siv2r) [reply reference.py]: Fixed

**Comment by siv2r** [comment]
Rebased. Also, added the `tests.sh` file

**Review by jonasnick — APPROVED** [review]
ACK 4885e36b1e06de82ab931b5eed7d2e9d320a334b

**Pull request merged**